### ISSUE TITLE
[feature request] Add support for 'checkbox' type menu item in fake context menu

### ISSUE THREAD

**Issue by toff** [piroor/treestyletab#1970]
Hi,
I have an extension that adds a tab context menu with a 'checkbox' type item.
I've tried TST's fake context menu API but it currently doesn't support this type of menu item.
In fact, it is already half working because the checkbox is displayed correctly.
I think there is 2 things missing:
1) Invert the 'checked' property.
2) Having a way to update the menu item before the menu is displayed like Firefox's new 'browser.menus.onShown' event.
I'm currently using 'tab-clicked' event but the menu is blinking when it is updated.

Thanks.

**Comment by piroor** [comment]
OK, these features have been implemented by recent commits. I've updated the API doc:
https://github.com/piroor/treestyletab/wiki/API-for-other-addons#add-new-item-to-the-context-menu-on-tabs
https://github.com/piroor/treestyletab/wiki/API-for-other-addons#when-the-context-menu-is-opened-and-closed

I've tested these new features with simple test cases, but there may have unintentional bugs. I'm waiting for your feedback.
